### DISCUSSION TITLE
How to create a schema extended from `record`?

### DISCUSSION THREAD

**Discussion by kostysh** [colinhacks/zod#2117] [Q&A] [answered]
Hi, in Typescript it is possible to do the following:

```typescript
type One = Record<string, string>;
interface Two extends Partial<One> {
  prop: string;
}
/*
interface Two {
  prop: string;
  [k: string]?: string;
}
*/
```

In Zod, the `record` schema does not have an `extend` feature.
How to implement the case mentioned above in Zod? Is it possible?

**Comment by JacobWeisenburger** [comment, marked as answer]
Is this what you are looking for?

```ts
const schema = z.object( {
    prop: z.number(),
} ).catchall( z.string() )
type Data = z.infer<typeof schema>
// type Data = {
//     [ x: string ]: string
//     prop: number
// }
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger
